x-pack/legacy/plugins/siem/public/lib/settings/use_kibana_ui_setting.ts

```diff
@@ -31,8 +31,9 @@ type GenericValue = string | boolean | number;
  * because the underlying `UiSettingsClient` doesn't support that.
  */
 export const useKibanaUiSetting = (key: string, defaultValue?: GenericValue) => {
```

Comment:
Not necessary for NP migration, but you could adopt some of the utilities from `kibana_react` instead of maintaining your own https://github.com/elastic/kibana/blob/master/src/plugins/kibana_react/README.md